battles/forms.py

```diff
@@ -18,48 +19,78 @@ def __init__(self, *args, **kwargs):
         self.fields['opponent'].queryset = users
 
 
+def validate_integer_doesnt_start_with_zero(value):
+    if str(value).startswith('0'):
+        raise forms.ValidationError(
+            'Your Pokemon id ({pokemon_id}) cannot start with 0.'.format(
+                pokemon_id=value
+            )
+        )
+
+
 class ChooseTeamForm(forms.ModelForm):
     class Meta:
         model = BattleTeam
         fields = []
 
     first_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='First pokemon')
+        min_value=1, max_value=802, required=True, label='First pokemon',
+        validators=[validate_integer_doesnt_start_with_zero])
     second_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='Second pokemon')
+        min_value=1, max_value=802, required=True, label='Second pokemon',
+        validators=[validate_integer_doesnt_start_with_zero])
     third_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='Third pokemon')
+        min_value=1, max_value=802, required=True, label='Third pokemon',
+        validators=[validate_integer_doesnt_start_with_zero])
+
+    def clean_first_pokemon(self):
+        value = self.cleaned_data.get('first_pokemon')
+        pokemon = init_pokemon_object(value)
+        return pokemon
+
+    def clean_second_pokemon(self):
+        value = self.cleaned_data.get('second_pokemon')
+        pokemon = init_pokemon_object(value)
+        return pokemon
+
+    def clean_third_pokemon(self):
+        value = self.cleaned_data.get('third_pokemon')
+        pokemon = init_pokemon_object(value)
```

Comment:
The actual purpose of `init_pokemon_object` is to check if the Pokemon exist and, if it doesn't, initialize their object, but not saving it yet. We're saving only after the validations, because saving in the helper would probably create Pokemon unnecessarily (that would answer your question about not doing so directly in the helper).

Does it make sense?

Reply:
It does! I have a suggestion for this: Transform the `init_pokemon_object` on a `get_or_create_pokemon` (https://docs.djangoproject.com/en/2.0/ref/models/querysets/#get-or-create) and use it only on the `save`. Doing this on the save would only save after the validation, would avoid calling the `init_pokemon_object` on every `clean_<field>` and would make the code clear.

Let me know your opinion :cat2:

Reply:
I've talked with Anderson and Flávio to understand better how I would do that and they told me it's okay to use it like this for now, since I'm matching most of my information with the API.

I had some difficulties trying to understand how I would use the `defaults` since pretty much all of my Pokemon's info are distinct and to use my instance I'd have to transform the returned object from `init_pokemon_object` into a dict and that would break some of the tests. But it does make sense, and I plan on refactoring that in the future.

Is that okay? :smiley_cat:

Reply:
Sure! No problem at all! :cat2: